Framework/Algorithms/src/DiscusMultipleScatteringCorrection.cpp

```diff
@@ -388,27 +427,21 @@ void DiscusMultipleScatteringCorrection::exec() {
   }
 }
 
-/**
- * Calculate a cumulative probability distribution for use in importance sampling. The distribution
- * is the inverse function P^-1(t4) where P(Q) = I(Q)/I(2k) and I(x) = integral of Q.S(Q)dQ between 0 and x
- * @param kinc The incident wavenumber
- * @return the inverted cumulative probability distribution
- */
-std::shared_ptr<Mantid::HistogramData::Histogram>
-DiscusMultipleScatteringCorrection::prepareCumulativeProbForQ(double kinc) {
-
+std::unique_ptr<Mantid::HistogramData::Histogram> DiscusMultipleScatteringCorrection::prepareQSQ(double kmax) {
   std::vector<double> qValues = m_SQHist->readX();
   std::vector<double> SQValues = m_SQHist->readY();
   // add terminating points at 0 and 2k before multiplying by Q so no extrapolation problems
   if (qValues.front() > 0.) {
     qValues.insert(qValues.begin(), 0.);
     SQValues.insert(SQValues.begin(), SQValues.front());
   }
-  if (qValues.back() < 2 * kinc) {
-    qValues.push_back(2 * kinc);
-    SQValues.push_back(SQValues.back());
+  if (qValues.back() < 2 * kmax) {
```

Comment:
It doesn't look like there's a check for the last element being greater than kmax - is that not possible?

Reply:
The qValues vector needs to extend at least as far as 2\*kmax for some later processing to work (the later processing is creation of a cumulative probability distribution). So if there is a value in qValues that is greater than 2*kmax that's fine, it just means we don't need to add an extra point in here.
Re the variable names - the user supplies two input workspaces: a) an S(Q) profile and b) an input sample workspace that defines the axes over which the multiple scattering calculation is to be calculated. "kmax" here is the largest k value in the input sample workspace while qValues is the list of q values in the S(Q) profile. Perhaps I'll just add a doxygen documentation comment to this function to explain because I'm struggling to come up with a snappy variable name. Does that sound OK?

Reply:
Ah OK I see, if you are making changes in the file already perhaps a comment/doc string would be helpful, if you don't mind